frontend/model/contracts/chatroom.js

```diff
@@ -136,6 +141,66 @@ function emitMessageEvent ({ contractID, hash }: {
   sbp('okTurtles.events/emit', `${CHATROOM_MESSAGE_ACTION}-${contractID}`, { hash })
 }
 
+export function makeMentionFromUsername (username: string): {
+  me: string, all: string
+} {
+  return {
+    me: `@${username}`,
+    all: '@all'
+  }
+}
+
+function addMention ({ contractID, messageId, datetime, text, username, chatRoomName }: {
+  contractID: string,
+  messageId: string,
+  datetime: string,
+  text: string,
+  username: string,
+  chatRoomName: string
+}): void {
+  /**
+   * If 'READY_TO_JOIN_CHATROOM' is false, it means not syncing chatroom
+  */
+  if (sbp('okTurtles.data/get', 'READY_TO_JOIN_CHATROOM')) {
+    return
+  }
+
+  sbp('state/vuex/commit', 'addChatRoomUnreadMention', {
+    chatRoomId: contractID,
+    messageId,
+    createdDate: new Date(datetime).getTime()
+  })
+
+  const rootGetters = sbp('state/vuex/getters')
+  const groupID = rootGetters.groupIdFromChatRoomId(contractID)
+  makeNotification({
+    title: `# ${chatRoomName}`,
+    body: text,
+    icon: rootGetters.globalProfile2(groupID, username).picture
+  })
+
+  sbp('okTurtles.events/emit', MESSAGE_RECEIVE)
+}
+
+function deleteMention ({ contractID, messageId }: {
+  contractID: string, messageId: string
+}): void {
+  sbp('state/vuex/commit', 'deleteChatRoomUnreadMention', { chatRoomId: contractID, messageId })
+
+  // Commented to delete notification for mentions
+  // const rootState = sbp('state/vuex/state')
+  // if (!rootState.chatRoomUnread[contractID] ||
+  //   !rootState.chatRoomUnread[contractID].mentions.length) {
+  //   const rootGetters = sbp('state/vuex/getters')
+  //   const notification = rootGetters.currentGroupNotifications.find(n =>
+  //     n.type === 'MENTION_ADDED' && n.linkTo === `/group-chat/${contractID}`
+  //   )
+  //   if (notification) {
+  //     sbp('gi.notifications/remove', notification)
+  //   }
+  // }
+}
+
```

Comment:
Since I don't think we'll ever have chatroom mentions appear under the bell notifications menu, it's probably safe to delete this.